app/forms/qualification_details_form.rb

```diff
@@ -0,0 +1,143 @@
+
+  validates :qualifications_details_check,
+    inclusion: {
+      in: [true, false],
+      message: ->(form, _) { form.i18n_errors_path("qualifications_details_check") }
+    }
+
+  def initialize(claim:, journey:, params:)
+    super
+
+    self.qualifications_details_check = permitted_params.fetch(
+      :qualifications_details_check,
+      claim.qualifications_details_check
+    )
+  end
+
+  def save
+    return false unless valid?
+
+    claim.assign_attributes(
+      qualifications_details_check: qualifications_details_check
+    )
+
+    if qualifications_details_check
+      # Teacher has confirmed the details in the dqt record are correct, update
+      # the eligibility with these details
+      claim.claims.each { |c| set_qualifications_from_dqt_record(c.eligibility) }
+    else
+      # Teacher has said the details don't match what they expected so
+      # nullify them
+      claim.claims.each { |c| set_nil_qualifications(c.eligibility) }
+    end
+
+    claim.save!
+  end
+
+  def dqt_route_into_teaching
+    claim.dqt_teacher_record.route_into_teaching
+  end
+
+  def dqt_academic_date
+    AcademicYear.for(claim.dqt_teacher_record.academic_date)
+  end
+
+  # Often the DQT record will represent subject names in all lowercase
+  def dqt_itt_subjects
+    claim.dqt_teacher_record.itt_subjects.map do |subject|
+      (subject.downcase == subject) ? subject.titleize : subject
+    end.join(", ")
+  end
+
+  def show_degree_subjects?
+    claim.claims.any? do |c|
+      c.dqt_teacher_record.eligible_itt_subject_for_claim == :none_of_the_above
+    end && claim.dqt_teacher_record.degree_names.any?
+  end
+
+  def dqt_degree_subjects
+    claim.dqt_teacher_record.degree_names.map do |subject|
+      (subject.downcase == subject) ? subject.titleize : subject
+    end.join(", ")
+  end
+
+  def dqt_qts_award_date
+    AcademicYear.for(claim.dqt_teacher_record.qts_award_date)
+  end
+
+  private
+
+  def set_qualifications_from_dqt_record(eligibility)
+    dqt_record = eligibility.claim.dqt_teacher_record
+
+    case eligibility
+    when Policies::EarlyCareerPayments::Eligibility
+      eligibility.assign_attributes(
+        itt_academic_year: itt_academic_year(dqt_record, eligibility),
+        eligible_itt_subject: eligible_itt_subject(dqt_record, eligibility),
+        qualification: qualification(dqt_record, eligibility)
+      )
+    when Policies::LevellingUpPremiumPayments::Eligibility
+      eligibility.assign_attributes(
+        itt_academic_year: itt_academic_year(dqt_record, eligibility),
+        eligible_itt_subject: eligible_itt_subject(dqt_record, eligibility),
+        qualification: qualification(dqt_record, eligibility),
+        eligible_degree_subject: eligible_degree_subject(dqt_record, eligibility)
+      )
+    when Policies::StudentLoans::Eligibility
+      eligibility.qts_award_year = qts_award_year(dqt_record, eligibility)
+    else
+      fail "Unknown eligibility type #{eligibility.class}"
+    end
```

Comment:
I feel a bit conflicted here; it feels like there should be at least two different forms behaving differently here, one per journey, and they should be namespaced (?) 🤔 🤷 Thoughts?

Reply:
Yeah that makes sense, will split this into separate forms

Reply:
The only other option I can think of is to delegate this pre-filling to something else; I can see pros and cons for both solutions, though... I am also a bit concerned that the pre-filling, in general, is a way of bypassing model validation rules that do exist if we were to input those attributes manually (when/if the slugs -forms- are presented). Raising this for consideration. The older part of me is more concerned about data consistency than coupling, but I don't see this a blocker

Reply:
Yeah that works much better as separate form objects 👍